src/frontend/serializable_models/persistable.ts

```diff
@@ -0,0 +1,81 @@
+import { Serializable } from './serializable';
+
+/**
+ * Persistable is an interface for objects to be stored in local storage.
+ *
+ * @class Persistable
+ */
+export abstract class Persistable<T> implements Serializable<T> {
+    documentTitle: string;
+
+    /**
+     * constructor creates a new Persistable concretion object.
+     *
+     * @class State
+     * @constructor
+     * @param {string} title The key associated with the localstorage object
+     *   persisted.
+     */
+    constructor(title?: string) {
+        this.documentTitle = title || '';
+    }
+
+    /**
+     * persist persists the serialized object representation to local storage.
+     *
+     * @class Persistable
+     * @method persist
+     */
+    persist() {
+        localStorage.setItem(this.getDocumentTitle(), this.serialize());
+    }
+
+    /**
+     * Retrieves an JSON serialized object from local storage.
+     *
+     * @class Persistable
+     * @method retrieve
+     */
+    retrieve() {
+        const savedState = localStorage.getItem(this.getDocumentTitle()) || '{}';
+        this.deserialize(savedState);
+    }
+
+    /**
+     * empty checks if the serialized object has not yet been persisted.
+     *
+     * @class Persistable
+     * @method empty
+     * @returns {boolean}
+     */
+    empty(): boolean {
+        return !localStorage.getItem(this.getDocumentTitle());
+    }
+
+    // Takes the object and serialize it to a JSON string representation.
+    abstract serialize(): string;
+
+    // Takes the a JSON string representation and updates the properties of the
+    // object instance.
+    abstract deserialize(serialized: string);
+
+    // Returns a representation of the object as a TypeScript type T.
+    abstract asBakedType(): T;
```

Comment:
I think "baked" is maybe the wrong term here, since it's usually used to mean the fixed, unmodifiable representation of the original, but here I think we're doing the opposite (the serialized string would be the "baked" version.) Maybe we want this to be called `getUnderlyingObject` or something like that? Same comment for the `Serializable` interface, and maybe `BakedState` can be called something like `RawState`?